File: src/main/java/com/example/hoteljavafinal/gestionChambrev1/chambre/ChambreService.java
package com.example.hoteljavafinal.gestionChambrev1.chambre;

import com.example.hoteljavafinal.gestionChambrev1.chambre.ChambreEntity;
import com.example.hoteljavafinal.gestionChambrev1.chambre.ChambreRepository;
import com.example.hoteljavafinal.gestionChambrev1.chambre.exceptions.ChambreNotFoundException;
import jakarta.transaction.Transactional;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.UUID;

@Service
@Transactional
@AllArgsConstructor
public class ChambreService {

    private ChambreRepository chambreRepository;
    public List<ChambreEntity> getAllChambres() {
        List<ChambreEntity> chambresEntities = chambreRepository.findAll();
        return chambresEntities;
    }
    public ChambreEntity getChambreById(UUID id) throws ChambreNotFoundException {
        ChambreEntity chambre = chambreRepository.findById(id).orElseThrow(() -> new ChambreNotFoundException());
        return chambre;
    }

    public ChambreEntity saveChambre(ChambreEntity chambreEntity) {
        ChambreEntity chambre = chambreRepository.save(chambreEntity);
        return chambre;
    }
    public ChambreEntity updateChambre(ChambreEntity chambreEntity) {
        ChambreEntity chambre = chambreRepository.save(chambreEntity);
        return chambre;
    }

    public void deleteChambre(UUID id) {
        chambreRepository.deleteById(id);
    }
}

File: hotel_TP_CPP/gestionChambre/chambre/ChambreService.cpp
#include <iostream>
#include <vector>
#include <memory>
#include <pstore.h>
#include "ChambreEntity.h"
#include "ChambreRepository.h"

class ChambreService {
public:
    ChambreService(ChambreRepository& repo) : chambreRepository(repo) {}

    std::vector<ChambreEntity> getAllChambres() {
        return chambreRepository.findAll();
    }

    ChambreEntity getChambreById(const UUID& id) {
        auto chambreOpt = chambreRepository.findById(id);
        if (!chambreOpt) {
            throw std::runtime_error("ChambreNotFoundException");
        }
        return *chambreOpt;
    }

    ChambreEntity saveChambre(const ChambreEntity& chambreEntity) {
        return chambreRepository.save(chambreEntity);
    }

    ChambreEntity updateChambre(const ChambreEntity& chambreEntity) {
        return chambreRepository.save(chambreEntity);
    }

    void deleteChambre(const UUID& id) {
        chambreRepository.deleteById(id);
    }

private:
    ChambreRepository& chambreRepository;
};

int main() {
    DatabaseSession session; // Supposons une session de base de données
    ChambreRepository repo(session);
    ChambreService service(repo);

    // Exemple d'utilisation du service
    try {
        auto chambres = service.getAllChambres();
        for (const auto& chambre : chambres) {
            std::cout << "Chambre ID: " << chambre.getId() << std::endl;
        }

        auto chambre = service.getChambreById(UUID::generate());
        std::cout << "Chambre retrouvée: ID " << chambre.getId() << std::endl;

        // Ajouter, mettre à jour, supprimer des exemples ici
    } catch (const std::exception& e) {
        std::cout << "Exception caught: " << e.what() << std::endl;
    }

    return 0;
}
